src/utils/feed.ts

```diff
@@ -54,7 +55,14 @@ export class Feed {
   };
 
   static fromString = (str: string): Feed => {
-    const feed = JSON.parse(str);
-    return new Feed(feed);
+    var feed
+    try {
+      feed = JSON.parse(str);
+    } catch (err) {
+      console.log('err feed from string')
+      throw err
+    }
+
+    return new Feed(feed.title, feed.items);
```

Comment:
I see what you're doing here (although, I think this might've been working?)... however, this implementation will be lossy. There's additional metadata in a feed (see the object properties) that we'll lose in this implementation. Essentially we just want a feed serializer - JSON stored in the user's filesystem, loaded when we need it, updated and saved again.

Reply:
Yes totally lossy; this just seemed like the easiest way to get something that kind of works

Reply:
This is updated as a work in progress. The constructor takes a full JSON version of a feed, however, I wasn't sure how to do typescript in this case -- https://github.com/nichoth/blog/blob/photo/src/utils/feed.ts#L50

Reply:
I fixed this so that there is no longer a lossy constructor. However, in the process I made a new interface -- https://github.com/nichoth/blog/blob/photo/src/utils/feed.ts#L27 -- which feels a little redundant, but as far as i know interfaces are only a compile-time part of typescript, and should completely be removed from the compiled JS